include/dali/core/cuda_utils.h

```diff
@@ -49,10 +49,54 @@ __device__ inline float16 StaticCastGpu(float val) {
 }
 #endif  // defined(CUDART_VERSION) && CUDART_VERSION < 9000
 
-// Starts profiling DALI
-inline void DALIProfilerStart() { cudaProfilerStart(); }
+// allocator using device-side malloc/free
 
-inline void DALIProfilerStop() { cudaProfilerStop(); }
+template <typename T>
+struct device_side_allocator {
+  static __device__ T *allocate(size_t count) {
+    return static_cast<T *>(malloc(count * sizeof(T)));
```

Comment:
shouldn't it be cudaMalloc for GPU?

Reply:
No, it's device-side allocator. On device, there are `malloc` and `free` functions which allocate global memory unusable outside device code. Unlike memory allocated with `cudaMalloc`, this memory cannot be used by runtime APIs, such as `cudaMemcpy` and cannot be freed by host code. The amount of this memory is very limited, too - 8MB by default, AFAIK.

Reply:
Got it, thanks for the explanation!